### ISSUE TITLE
NVIDIA DLSS (Deep Learning Super Sampling)

### ISSUE THREAD

**Issue by JMS55** [bevyengine/bevy#8420]
## What problem does this solve or what need does it fill?

We already have TAA. DLSS takes temporal rendering a step further, and renders at a lower resolution and then upscales it over time. This can be a huge performance increase when the cost of shading a pixel is very expensive, such as in per-pixel ray/path tracing.

## What solution would you like?

Integrate DLSS as an optional anti-aliasing method, completing our existing MSAA, FXAA, TAA options.

## Current Progress
* I have a seemingly-working `dlss_wgpu` crate allowing usage of DLSS from wgpu https://github.com/JMS55/dlss_wgpu
  * Depends on my wgpu fork https://github.com/gfx-rs/wgpu/pull/3376
* I have a branch of bevy adding support for DLSS https://github.com/JMS55/bevy/blob/dlss/crates/bevy_core_pipeline/src/dlss.rs
  * Quality seems bad/wrong - there's heavy aliasing even at 4k quality mode. I'm not sure why.
  * There's a crash when enabling one of the ultra quality/performance modes.
  * I haven't setup support (in either dlss_wgpu/bevy) for DLSS A-F presets (different from the quality/performance modes).

## What alternative(s) have you considered?

DLSS is NVIDIA only, and closed-source. FSR (https://github.com/bevyengine/bevy/issues/5232) is an alternative option from AMD that works on more GPUs and is open-source. There's also XeSS from Intel.

I've looked into integrating FSR but ran into difficulties. I haven't bothered looking into XeSS.

**Comment by alice-i-cecile** [comment]
I'd prefer to prioritize a cross-hardware compatible algorithm in Bevy itself: we should just make sure that a third-party DLSS crate can interop cleanly.

**Comment by JMS55** [comment]
> I'd prefer to prioritize a cross-hardware compatible algorithm in Bevy itself: we should just make sure that a third-party DLSS crate can interop cleanly.

Me too, but I couldn't get FSR working. It would require deep wgpu changes around texture state tracking in order to work. That, or maybe a custom fork of FSR. I'm not sure what the solution would be. That's why I've shifted focus towards DLSS.

That said, I'd like to support _all_ options in bevy, out of the box. In the future we could have a `TemporalBundle` that tries DLSS, FSR, XeSS, MTLFXTemporalScaler, TAA, choosing the best one or falling back if not supported on the current platform.

**Comment by PROMETHIA-27** [comment]
I agree with alice; I think we should do what is necessary to ensure DLSS is not as easy as possible to integrate, but it should be strictly up to a third party to do so.

**Comment by PROMETHIA-27** [comment]
Yep, typo

**Comment by JMS55** [comment]
I have this basically finished in https://github.com/JMS55/bevy/tree/dlss2.

Blocked on wgpu v25 releasing.

In the mean time, there's a bunch of pieces we can extract and put into smaller PRs.

**Comment by jordanhalase** [comment]
#19563 upgrades wgpu to v25.0 in bevy

**Comment by madmaxio** [comment]
Is DLSS Frame Generation also in scope of this implementation?

**Comment by JMS55** [comment]
It's not something I'm personally interested in doing, but someone else could add support for it.
